app/jobs/regular/grant_badge.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Jobs
+  class GrantBadge < ::Jobs::Base
+    def execute(args)
+      return unless SiteSetting.enable_badges
+
+      badge = Badge.enabled.find_by(id: args[:badge_id])
+      return unless badge
+
+      # Cancels the scheduled job to ensure badge consistency as the badges are
+      # mutating during `BadgeGranter.backfill`.
+      Jobs.cancel_scheduled_job(:ensure_badge_consistency)
```

Comment:
Is `Jobs:EnsureBadgeConsistency` supposed to be a scheduled job because it is written as a regular job at the moment.